horace_core/algorithms/private/gen_sqw_check_files.m

```diff
@@ -155,26 +234,26 @@
 %See above (RAE)
 
 
-% Check that spe, par and sqw file names do not match
+% Check that spe, par and sqw file names do not clash
 % ---------------------------------------------------
 if any(strcmpi(sqw_file_out,spe_file_out)) && ~isempty(sqw_file_out)
     error('HORACE:algorithms:invalid_argument',...
         'Output sqw file name %s matches one of the input spe file names',sqw_file_out);
 end
 
-if ~isempty(par_file_out)
-    if any(strcmpi(par_file_out,spe_file_out))
-        error('HORACE:algorithms:invalid_argument',...
-            'Detector parameter file name %s matches one of the input spe file names',par_file_out);
-    elseif strcmpi(par_file_out,sqw_file_out)
-        error('HORACE:algorithms:invalid_argument',...
-            'Detector parameter file name %s and output sqw file name match',par_file_out);
+
+for ii=1:numel(par_file_out)
+    if ~isempty(par_file_names{ii})
+        for jj=1:numel(spe_file_out)
+            if any(strcmpi(par_file_names{ii},spe_file_out{jj}))
+                error('HORACE:algorithms:invalid_argument',...
+                    'Detector parameter file name %s matches one of the input spe file names',par_file_names{ii});
+            end
+        end
```

Comment:
Are these nested cell arrays? If not, you can get away with:
```suggestion
        if any(strcmpi(par_file_names{ii}, spe_file_out))
            error('HORACE:algorithms:invalid_argument',...
                 'Detector parameter file name %s matches one of the input spe file names',par_file_names{ii});
        end
```

Reply:
worth knowing but if the present one works, leaving